packages/aws_signature_v4/test/c_test_suite/test_data_loader.dart

```diff
@@ -0,0 +1,193 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// ignore_for_file: avoid_print
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+
+import 'context.dart';
+import 'request_parser.dart';
+import 'test_data.dart';
+
+/// Handles parsing for one file.
+typedef TestFileParser = Future<void> Function(SignerTestBuilder, String);
+
+/// File parsers for the current test suite directory organization.
+const Map<String, TestFileParser> _fileParsers = {
+  'context.json': _parseContext,
+  'request.txt': _parseRequest,
+  'query-canonical-request.txt': _parseQueryCanonicalRequest,
+  'query-string-to-sign.txt': _parseQueryStringToSign,
+  'query-signature.txt': _parseQuerySignature,
+  'query-signed-request.txt': _parseQuerySignedRequest,
+  'header-canonical-request.txt': _parseHeaderCanonicalRequest,
+  'header-string-to-sign.txt': _parseHeaderStringToSign,
+  'header-signature.txt': _parseHeaderSignature,
+  'header-signed-request.txt': _parseHeaderSignedRequest,
+};
+
+/// Parses raw HTTP requests.
+final _requestParser = SignerRequestParser();
+
+/// Loads all test cases in the C signer test suite by walking the file tree
+/// and creating a [SignerTest] for each folder encountered where all requisite
+/// files are available.
+Future<List<SignerTest>> loadAllTests() async {
+  await _requestParser.init();
+
+  final testSuitePath = path.joinAll([
+    Directory.current.path,
+    'external',
+    'aws-c-auth',
+    'tests',
+    'aws-signing-test-suite',
+    'v4',
+  ]);
+  final testCases = <SignerTest>[];
+
+  for (var entity in Directory(testSuitePath).listSync()) {
+    final stat = entity.statSync();
+    if (stat.type != FileSystemEntityType.directory) {
+      throw StateError('Unexpected top level file found: ${entity.path}');
+    }
+    final testCaseDir = Directory(entity.path);
+    final testCaseName = path.basename(entity.path);
+    final testCaseBuilder = SignerTestBuilder(testCaseName);
+
+    // List all the files in the test directory and sort them alphabetically.
+    // This is required to ensure that `context.json` is the first file parsed,
+    // since it's data is used to parse other files accordingly, namely `request.txt`.
```

Comment:
```suggestion
    // This is required to ensure that `context.json` is the first file parsed
    // since its data is used to parse other files accordingly, namely `request.txt`.
```
Could you also help me understand why this ensures context.json is the first file parsed?

Reply:
Sure, it's because the directory layout is the same for each test.

<img width="298" alt="Screen Shot 2022-04-19 at 5 16 27 PM" src="https://user-images.githubusercontent.com/24740863/164121640-e87e897d-3239-4bc1-8552-1bc47c014210.png">

Reply:
Same reply as below 🤣  - it's definitely good to know that we at least have manual update as a friction point but if we were all replaced tomorrow by a whole new team of engineers, it would just have been nice not to set them up for failure. But yeah, not a huge concern by any means

Reply:
Their response:
> Those were probably in the original test suite we started with. There were several tests that had flat-out wrong validation data in them and we skipped. We should remove these.

Updated with list of ignored tests and made it throw now if any file is missing